application/models/Calculator.php

```diff
@@ -68,4 +68,18 @@ public function multiply()
     {
         return $this->firstNumber * $this->secondNumber;
     }
+
+    /**
+     * Calculate the division of two numbers.
+     * @return float|int
+     * @throws \Exception
+     */
+    public function divide()
+    {
+        if ($this->secondNumber === 0) {
+            throw new  \Exception('Divide by zero is not allowed');
+        }
+
+        return (float) $this->firstNumber / $this->secondNumber;

```

Comment:
```suggestion
        return fmod($this->firstNumber, $this->secondNumber);

```
[](https://www.w3schools.com/php/func_math_fmod.asp)

Reply:
```
>>> fmod(34,3)
=> 1.0
>>> 34/3
=> 11.333333333333

```